zendesk/resource_zendesk_group_test.go

```diff
@@ -63,25 +124,82 @@ func TestCreateGroup(t *testing.T) {
 	}
 }
 
-func TestUnmarshalGroup(t *testing.T) {
-	m := &identifiableMapGetterSetter{
-		id: "1234",
-		mapGetterSetter: mapGetterSetter{
-			"url":  "https://example.zendesk.com/api/v2/ticket_fields/360011737434.json",
-			"name": "name",
-		},
+func TestUpdateGroup(t *testing.T) {
+	ctrl := NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id:              "12345",
+		mapGetterSetter: make(mapGetterSetter),
 	}
 
-	g, err := unmarshalGroup(m)
-	if err != nil {
-		t.Fatalf("Could marshal map %v", err)
+	m.EXPECT().UpdateGroup(Eq(int64(12345)), Any()).Return(zendesk.Group{}, nil)
+	if err := updateGroup(i, m); err != nil {
+		t.Fatal("readGroup returned an error")
 	}
+}
 
-	if v := m.Get("url"); g.URL != v {
-		t.Fatalf("group had url value %v. should have been %v", g.URL, v)
+func TestDeleteGroup(t *testing.T) {
+	ctrl := NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id: "12345",
 	}
 
-	if v := m.Get("name"); g.Name != v {
-		t.Fatalf("group had name value %v. should have been %v", g.Name, v)
+	m.EXPECT().DeleteGroup(Eq(int64(12345))).Return(nil)
+	if err := deleteGroup(i, m); err != nil {
+		t.Fatal("readGroup returned an error")
+	}
+}
+
+func testGroupDestroyed(s *terraform.State) error {
+	client := testAccProvider.Meta().(zendesk.GroupAPI)
+
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "zendesk_group" {
+			continue
+		}
+
+		id, err := atoi64(r.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = client.GetGroup(id)
+		if err == nil {
+			return fmt.Errorf("did not get error from zendesk when trying to fetch the destroyed group")
+		}
+
+		zd, ok := err.(zendesk.Error)
+		if !ok {
+			return fmt.Errorf("error %v cannot be asserted as a zendesk error", err)
+		}
+
+		if zd.Status() != http.StatusNotFound {
+			return fmt.Errorf("did not get a not found error after destroy. error was %v", zd)
+		}
 	}
+
+	return nil
+}
+
+func TestAccGroupExample(t *testing.T) {
```

Comment:
when i ran this locally i got the following error.
```
--- FAIL: TestAccGroupExample (1.23s)
    testing.go:599: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: Check failed: did not get error from zendesk when trying to fetch the destroyed group
        
        State: <no state>

```

Reply:
Looking at the groups that do exist in zendesk it may be happening bc or this group name being used for the default group. https://github.com/nukosuke/terraform-provider-zendesk/blob/master/examples/groups.tf#L8 Might want to consider changing that.

Reply:
Changed the name from `Support` to `Moderator`.